src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreRT.cs

```diff
@@ -136,6 +137,7 @@ public static unsafe void DestroyStructure(IntPtr ptr, Type structuretype)
             }
         }
 
+        [RequiresDynamicCode("Marshalling code for the object might not be available")]
```

Comment:
Should we include suggestion for how to fix where possible? E.g. "Use Marshal.StructurePtr<T> instead."